iml-gui/crate/src/components/action_dropdown/confirm_action_modal.rs

```diff
@@ -0,0 +1,209 @@
+use crate::{
+    components::{
+        action_dropdown::{state_change, DryRun},
+        font_awesome, modal,
+    },
+    extensions::{MergeAttrs, NodeExt},
+    generated::css_classes::C,
+    key_codes, GMsg, RequestExt,
+};
+use iml_wire_types::{warp_drive::ErasedRecord, AvailableAction, Command, EndpointName};
+use seed::{prelude::*, *};
+use std::sync::Arc;
+
+#[derive(serde::Serialize)]
+pub struct SendJob<'a, T> {
+    pub class_name: &'a Option<String>,
+    pub args: &'a T,
+}
+
+#[derive(serde::Serialize)]
+pub struct SendCmd<'a, T> {
+    pub jobs: Vec<SendJob<'a, T>>,
+    pub message: String,
+}
+
+#[derive(Default)]
+pub struct Model {
+    pub modal: modal::Model,
+}
+
+#[derive(Clone)]
+pub enum Msg {
+    SendJob(String, Arc<AvailableAction>),
+    JobSent(Box<fetch::ResponseDataResult<Command>>),
+    SendStateChange(Arc<AvailableAction>, Arc<dyn ErasedRecord>),
+    StateChangeSent(Box<fetch::ResponseDataResult<Command>>),
+    Modal(modal::Msg),
+    Noop,
+}
+
+pub enum Action {
+    Loading,
+    Job(String, Arc<AvailableAction>, Arc<dyn ErasedRecord>),
+    StateChange(DryRun, Arc<AvailableAction>, Arc<dyn ErasedRecord>),
+}
+
+pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg, GMsg>) {
+    match msg {
+        Msg::SendJob(message, action) => {
+            let x = SendCmd {
+                jobs: vec![SendJob {
+                    class_name: &action.class_name,
+                    args: &action.args,
+                }],
+                message,
+            };
+
+            let req = fetch::Request::api_call(Command::endpoint_name())
+                .with_auth()
+                .method(fetch::Method::Post)
+                .send_json(&x);
+
+            orders
+                .perform_cmd(req.fetch_json_data(|x| Msg::JobSent(Box::new(x))))
+                .send_msg(Msg::Modal(modal::Msg::Close));
+        }
+        Msg::JobSent(_) => {
+            //@TODO: Open command modal here
+        }
+        Msg::SendStateChange(action, erased_record) => {
+            let req = state_change(&action, &erased_record, false);
+
+            orders
+                .perform_cmd(req.fetch_json_data(|x| Msg::StateChangeSent(Box::new(x))))
+                .send_msg(Msg::Modal(modal::Msg::Close));
+        }
+        Msg::StateChangeSent(_) => {
+            //@TODO: Open command modal here
+        }
+        Msg::Modal(msg) => {
+            modal::update(msg, &mut model.modal, &mut orders.proxy(Msg::Modal));
+        }
+        Msg::Noop => {}
+    };
+}
+
+pub(crate) fn view(action: &Action) -> Node<Msg> {
+    let confirm_msg = match action {
+        Action::Loading => Msg::Noop,
+        Action::Job(_, action, erased_record) => {
```

Comment:
Nit: It might be just me, but "erased" sounds confusing to me, maybe `some_record`?